Solopov/DZ5_AjaxAsynchronousMethodAndCSS/ServletTestList/src/main/java/services/UserService.java

```diff
@@ -1,41 +1,41 @@
 package services;
 
-import entity.User;
+import hibernate.entity.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
 
-import java.util.LinkedHashMap;
-import java.util.Map;
+import java.util.*;
 
+@Service
 public class UserService {
-private static Map<String,String> account =new LinkedHashMap<>();//All registered users. key -login, value -password
-    private User currentUser;
-
-    public boolean addAccount(String login, String password){
-        if (!account.containsKey(login)){
-            account.put(login,password);
-            User currentUser=new User(login,password);
-            return true;
+    @Autowired
+    private hibernate.service.ServiceUser userServ;
+    private static User currentUser;
+    private static NavigableSet<User> allUsers= new TreeSet<>();
+
+
+    public boolean addAccount(User user) {
+        allUsers = new TreeSet<>(userServ.findAll());
+        boolean result= allUsers.add(user);
+        if(result) {
+            userServ.persist(user);
+            currentUser = user;
         }
-        return false;
-
-    }
-    public boolean authorizate(String log,String pass){
-        String passValue= account.get(log);
-       if(pass.equals(passValue)) return true;
-        return false;
+            return result;
     }
 
-
-
-
-    public static Map<String, String> getAccount() {
-        return account;
+    public boolean authorizate(String login, String pass) {
+        allUsers = new TreeSet<>(userServ.findAll());
+        currentUser = userServ.findByLogin(login);
+        if (currentUser!= null && pass.equals(currentUser.getPassword())) return true;
+        return false;
     }
 
-    public User getCurrentUser() {
+    public static User getCurrentUser() {
```

Comment:
Static field?

Reply:
Yes, this is so as not to waste time extracting from the database constantly, but once. We extract one user and use it permanently. And it is more convenient to use a statics, you do not need to produce objects